### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by aksoli666**
+Added test cases for Book and Category entities 
(repository, service, controller).
Знімок екрана 2024-06-11 092207

### CODE REVIEW

File: src/main/java/online/book/store/config/SecurityConfig.java [hunk 1]
```diff
@@ -15,13 +16,15 @@
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 
+@EnableMethodSecurity
 @EnableWebSecurity
```
Comment (Elena-Bruyako) [inline src/main/java/online/book/store/config/SecurityConfig.java:22]: Do we need both of `@EnableMethodSecurity and @EnableWebSecurity` here?

File: src/main/java/online/book/store/repository/UserRepository.java [hunk 2]
```diff
@@ -3,9 +3,11 @@
 import java.util.Optional;
 import online.book.store.model.User;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface UserRepository extends JpaRepository<User, Integer> {
     boolean existsUserByEmail(String email);
 
+    @Query(value = "SELECT u FROM User u LEFT JOIN FETCH u.roles where u.email = :email")
```
Comment (Elena-Bruyako) [inline src/main/java/online/book/store/repository/UserRepository.java:11]:
```suggestion
    @EntityGraph(attributePaths = "roles")
```

File: src/main/java/online/book/store/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -23,8 +25,8 @@ public UserResponseDto register(UserRegistrationRequestDto userRequestDto) {
                     .format("User with this email: %s already exists",
                             userRequestDto.getEmail()));
         }
-        User user = new User();
-        userMapper.toModel(userRequestDto);
+        User user = userMapper.toModel(userRequestDto);
+        user.setPassword(passwordEncoder.encode(userRequestDto.getPassword()));
```
Comment (Elena-Bruyako) [inline src/main/java/online/book/store/service/impl/UserServiceImpl.java:29]:
```suggestion
        user.setPassword(passwordEncoder.encode(userRequestDto.getPassword()));
        user.setRole....
```

File: src/test/java/online/book/store/controller/BookControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+                MockMvcRequestBuilders.get(URL_TEMPLATE_WITHOUT_ID)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper.readValue(
+                result.getResponse()
+                        .getContentAsString(),
+                BookDto[].class);
+        Arrays.stream(actual).forEach(System.out::println);
```
Comment (Elena-Bruyako) [inline src/test/java/online/book/store/controller/BookControllerTest.java:225]: Do we need this?

File: src/test/java/online/book/store/service/BookServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+        bookDto.setId(VALID_BOOK_ID_1L);
+        bookDto.setTitle(book.getTitle());
+
+        Mockito.when(bookMapper.toModel(requestDto)).thenReturn(book);
+        Mockito.when(bookRepository.save(book)).thenReturn(book);
+        Mockito.when(bookMapper.toDto(book)).thenReturn(bookDto);
+
+        BookDto savedBookDto = bookService.save(requestDto);
+
+        Assertions.assertThat(savedBookDto).isEqualTo(bookDto);
```
Comment (Elena-Bruyako) [inline src/test/java/online/book/store/service/BookServiceTest.java:71]:
```suggestion
        assertEquals(bookDto, savedBookDto);
```

Comment (Elena-Bruyako) [reply src/test/java/online/book/store/service/BookServiceTest.java:71]: Check other places

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
good job

File: src/test/resources/database/online-book-store/book/add-book-for-update.sql [hunk 6]
```diff
@@ -0,0 +1 @@
+insert into books (id, title, author, isbn, price, description, cover_image, user_id) values (8, "Title8", "Author8", "isbn8", 58, "description8", "cover img8", 1);
```
Comment (Vasul-Pankiv) [inline src/test/resources/database/online-book-store/book/add-book-for-update.sql:1]: all files has to end with empty line

File: src/test/java/online/book/store/service/BookServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+        assertThrows(
+                EntityNotFoundException.class,
+                () -> bookService.findById(INVALID_BOOK_ID));
+    }
+
+    @Test
+    @DisplayName("Verify findAll() method, return List of BookDto")
+    public void findAll_ValidPageable_ReturnListBookDto() {
+        pageable = PageRequest.of(PAGE_NUMBER, PAGE_SIZE);
+        Book book1 = new Book();
```
Comment (Vasul-Pankiv) [inline src/test/java/online/book/store/service/BookServiceTest.java:109]: try to create BookProvider class with static method to get needed books

Comment (Vasul-Pankiv) [reply src/test/java/online/book/store/service/BookServiceTest.java:109]: it's up to you

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/test/resources/database/online-book-store/book/add-book-for-update.sql (hunk 6 above)
Comment (Elena-Bruyako) [reply src/test/resources/database/online-book-store/book/add-book-for-update.sql]: Not fixed

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Please, fix comment that mentor left

**Comment by aksoli666** [comment]
> Please, fix comment that mentor left
To create a BookProvider class was a wish, and in the files I added empty lines

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**